test/unit/services/synchronization/nowait_lock_service_test.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Synchronization::NowaitLockServiceTest < ActionDispatch::IntegrationTest
+  attr_reader :lock_key
+
+  setup do
+    @lock_key = SecureRandom.uuid
+  end
+
+  test "locks then cannot lock but can lock after unlock" do
+    assert Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
+    assert_not Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
+    assert_equal 1, Synchronization::UnsafeUnlockService.call(lock_key).result
+    assert_equal 0, Synchronization::UnsafeUnlockService.call(lock_key).result
+    assert Synchronization::NowaitLockService.call(lock_key, timeout: 100).result
+  end
+
+  test "locks between threads" do
```

Comment:
This waits for one thread to finish before the other thread tries to acquire the lock. Do you think there's a value on testing a race condition between threads? Something like:

```
    wait = true
    Thread.new do
      sleep 0.001 while wait
      Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
    end
    Thread.new do
      sleep 0.001 while wait
      Synchronization::NowaitLockService.call(lock_key, timeout: 10000).result
    end
    wait = false
```

Both threads try to get the same lock at the same time, one of them should take it, the other shouldn't. WDYT?

Reply:
This is again testing the underlying library and I doubt we can very well test it. It has tests.

https://github.com/leandromoreira/redlock-rb/tree/main

We also test billing under 3 concurrent threads here:
https://github.com/3scale/porta/pull/3403/files#diff-de91bc552e1d866b632914c797200e94d135101296faefc5eb70842ca02e04e4R27

To me it's over the top to try to test it more.

Reply:
I'm trying it out once manually to make sure configuration works as expected. But I will not write a test for it. It is too complicated. It is not meant to be testable as library internals.